src/main/java/com/minecolonies/coremod/colony/managers/ColonyPackageManager.java

```diff
@@ -153,116 +117,93 @@ else if (lastContactInHours != 0)
         }
         colony.getRequestManager().setDirty(false);
     }
 
     @Override
-    public void sendPermissionsPackets(@NotNull final Set<EntityPlayerMP> oldSubscribers, final boolean hasNewSubscribers)
+    public void sendPermissionsPackets()
     {
         final Permissions permissions = colony.getPermissions();
-        if (permissions.isDirty() || hasNewSubscribers || colony.getWorld().rand.nextInt(CHANCE_TO_UPDATE) <= 1)
+        if (permissions.isDirty() || !newSubscribers.isEmpty())
         {
-            subscribers
-                    .stream()
-                    .filter(player -> permissions.isDirty() || !oldSubscribers.contains(player)).forEach(player ->
-            {
-                final Rank rank = permissions.getRank(player);
-                MineColonies.getNetwork().sendTo(new PermissionsMessage.View(colony, rank), player);
-            });
+            final Set<EntityPlayerMP> players = permissions.isDirty() ? closeSubscribers : newSubscribers;
+            players.forEach(player -> MineColonies.getNetwork().sendTo(new PermissionsMessage.View(colony, permissions.getRank(player)), player));
         }
     }
 
     @Override
-    public void sendWorkOrderPackets(@NotNull final Set<EntityPlayerMP> oldSubscribers, final boolean hasNewSubscribers)
+    public void sendWorkOrderPackets()
     {
         final IWorkManager workManager = colony.getWorkManager();
-        if (workManager.isDirty() || hasNewSubscribers)
+        if (workManager.isDirty() || !newSubscribers.isEmpty())
         {
+            final Set<EntityPlayerMP> players = workManager.isDirty() ? closeSubscribers : newSubscribers;
             for (final IWorkOrder workOrder : workManager.getWorkOrders().values())
             {
                 if (!(workOrder instanceof WorkOrderBuildMiner))
                 {
-                    subscribers.stream().filter(player -> workManager.isDirty() || !oldSubscribers.contains(player))
-                            .forEach(player -> MineColonies.getNetwork().sendTo(new ColonyViewWorkOrderMessage(colony, workOrder), player));
+                    players.forEach(player -> MineColonies.getNetwork().sendTo(new ColonyViewWorkOrderMessage(colony, workOrder), player));
                 }
             }
-
             workManager.setDirty(false);
         }
     }
 
     @Override
-    public void sendSchematicsPackets(final boolean hasNewSubscribers)
+    public void sendSchematicsPackets()
     {
-        if (Structures.isDirty() || hasNewSubscribers)
+        if (Structures.isDirty() || !newSubscribers.isEmpty())
         {
-            subscribers.stream()
-                    .forEach(player -> MineColonies.getNetwork().sendTo(new ColonyStylesMessage(), player));
+            final Set<EntityPlayerMP> players = Structures.isDirty() ? closeSubscribers : newSubscribers;
```

Comment:
Each of these messages has 90% the same code, maybe we can do something about this?